rocketpy/utilities.py

```diff
@@ -685,3 +690,71 @@ def get_instance_attributes(instance):
         if not inspect.ismethod(member[1]) and not member[0].startswith("__"):
             attributes_dict[member[0]] = member[1]
     return attributes_dict
+
+
+def save_to_rpy(flight: Flight, filename: str, include_outputs = False):
+    """Saves a .rpy file into the given path, containing key simulation 
+    informations to reproduce the results.
+
+    Parameters
+    ----------
+    flight : rocketpy.Flight
+        Flight object containing the rocket's flight data
+    filename : str
+        Path where the file will be saved in
+    include_output : bool, optional
+        If True, the function will include extra outputs into the file,
+        by default False
+        
+    Returns
+    -------
+    None
+    """
+    file = os.path.splitext(filename)[0] + ".rpy"
```

Comment:
```from pathlib import Path``` 

```suggestion
    file = Path(filename).with_suffix(".rpy")

```

Reply:
Pathlib is more specialized to dealing with paths, and it is a part of the standard lib, so it's more recommended in this case.
I was afraid of possible errors from the `+` operation. Perhaps the first part may not always be a string.